### DISCUSSION TITLE
Rpc function created returns an array of users

### DISCUSSION THREAD

**Discussion by ricardosikic** [supabase/supabase#18480] [Questions] [answered]
Hello, I have created an RPC function to retrieve profile data for a user. This function is called from a frontend view, where the logged-in user's ID is passed as an argument. However, the function is currently returning data for all users in the application as they log in, instead of returning only the information for the requested user.

```
  const getProfile = async() => {
    const user = supabase.auth.user();
    let {data, error, status} = await supabase.rpc('get_profile_with_role', {user_id: user.id});
    if(error) throw error;
    if(data && data.length > 0) {
      let cleanUserName = data[0].username ? data[0].username.replace(/['"]+/g, '') : '';
      setUserInfo(data[0]);
    
    }
  }

```

The RPC looks like this.

```
CREATE OR REPLACE FUNCTION public.get_profile_with_role(user_id uuid)
 RETURNS TABLE(
  id uuid,
  username text,
  name text,
  website text, 
  avatar_url text, 
  welcome_email_sent boolean, 
  has_accepted_privacy_policy boolean,
  country text,
  phone_number text,
  about_me text,
  gender text,
  city text,
  rut text,
  role_id integer
 )
 LANGUAGE sql AS
$func$
SELECT 
  p.id,
  p.username,
  p.name,
  p.website,
  p.avatar_url,
  p.welcome_email_sent,
  p.has_accepted_privacy_policy,
  p.country,
  p.phone_number,
  p.about_me,
  p.city,
  p.rut,
  ur.role_id
FROM 
profiles p
JOIN user_roles ur ON ur.user_id = p.id
where p.id = user_id
$func$;
```

**Comment by GaryAustin1** [comment, marked as answer]
Hard to spot errors in postgres, but in general I avoid naming parameters the same as column names as sometimes that causes confusion when doing multiple table queries.   Speaking of user_id in your case. 

Also auth.user() is version 1 supabase-js which is about a year deprecated.

**Reply by ricardosikic** [reply]
Hey, it works. When the query was executed there was a column called `user_id`. And about the deprecated function you are right i have to upgrade the version on my project.
